### PR TITLE
Allow fingerprint to be explicitly specified.

### PR DESCRIPTION
**Pull request by petemounce**
This allows a CI pipeline to explicitly trust particular server fingerprint(s) when interacting over ssl.

See documentation links within README.md's diff.

## Changes

Add `trusted_fingerprint` setting to plugin.

## Verification

* [x] verify change works :)
    * this was verified internally on linux and Windows within game studio CI

### CODE REVIEW

**Comment by ca-johnson** [comment]
LGTM in principle, but there are better places to put this code than the hook script. Ideally these scripts are kept to the absolute minimum, where the only logic is stuff that cannot feasibly be executed in python (e.g, creating the virtualenv for python 🕺)

Param parsing can be moved to `buildkite.py:get_config()`, using existing util `list_from_env_array` (like `sync` param)

Then add another param to `perforce.py:P4Repo:__init__()` called `fingerprints`, and in line 55, if fingerprints are specified explicitly add those with `fingerprint -i -f` (and retain the past behaviour if not, for backwards compatibility)

As far as unit testing goes.. this is a bit more fiddly as I believe the fixture is not currently using ssl, but should be possible to enable without much difficulty.

Also worth mentioning: I prefer the name `fingerprints` vs `trusted_fingerprints` - the `trusted` part I think is clearly implied (do we expect to have any other variation of fingerprint? is anyone likely to make a mistake and provide the wrong thing here?)

This is actually quite a good change to pair on as its not too involved but touches quite a few important parts of the code.

**Comment by improbable-mattchurch** [comment]
hot damn this is a good change!

**Comment by ca-johnson** [comment]
Nice. Have added some feedback/functional improvement in https://github.com/improbable-eng/perforce-buildkite-plugin/pull/196

File: python/test_perforce.py [hunk 1]
```diff
@@ -82,12 +89,11 @@ def store_server(repo, to_zip):
                 abs_path = os.path.join(root, filename)
                 archive.write(abs_path, os.path.relpath(abs_path, serverRoot))
 
-def test_fixture(capsys, server):
+def test_server_fixture(capsys, server):
```
Comment (ca-johnson) [inline python/test_perforce.py:92]: There are some references this in README.md to update

**Review by ca-johnson — APPROVED** [review]
LGTM 👍

**Pull request merged**